eclipse-theia/theia
# access tasks configs as preferences

- In current Theia, the task extension is responsible for reading,
writing, and watching `tasks.json` files. With this change, the task
extension does not access `tasks.json` files directly, and leaves the
work to the preference extension.
- resolves #5013


#### How to test
There are no changes in terms of functionalities. Therefore, to verify this change, we would need to go through the use cases that are already supported by Theia, and  check if they are still properly functional. Those use cases include but not limit to:

As a Theia user I am able to

- run tasks defined in `tasks.json`
- run detected tasks contributed by Theia plugins
- run detected tasks contributed by Theia extensions (Please note, this is broken on master branch and reported in https://github.com/eclipse-theia/theia/issues/6204)
- configure tasks defined in `tasks.json`
- configure detected tasks (Please note, "configuring tasks" by clicking the gear icons in tasks' quick open menu is broken on master branch and reported in https://github.com/eclipse-theia/theia/issues/6212. In order to test this feature we would have to use the top menu bar `Terminal` -> `Configure Tasks ...`)
- see the recently used tasks be grouped together and show up at the top of the tasks' quick open menu
- see the "detected tasks" displayed as "configured tasks", if they are configured / customized in the `tasks.json`
- see the problem matcher items show up in the prompt, if a task that is not associated with any problem matchers is started
   - and see the selected problem matcher item written into `tasks.json` 
- do all the items described above in both single root and multi root workspace.
